lib/modules/commentNavigator.js

```diff
@@ -248,22 +237,37 @@ function onScroll() {
 	}
 }
 
+function enableControls() {
+	commentNav().up.disabled = false;
+	commentNav().down.disabled = false;
+	commentNav().buttons.classList.remove('noNav');
+}
+
+function disableControls() {
+	commentNav().up.disabled = true;
+	commentNav().down.disabled = true;
+	commentNav().buttons.classList.add('noNav');
+}
+
 function changeCategory() {
 	const index = commentNav().select.selectedIndex;
 	if (index === -1) {
 		return;
 	}
-	const newCategory = (commentNav().select.options.item(index).value: any);
-	if (newCategory !== currentCategory) {
-		resetNavigator(newCategory);
+	const previousCategory = currentCategory;
+	currentCategory = (commentNav().select.options.item(index).value: any);
+	refreshCurrentCategoryPosts();
+	if (previousCategory !== currentCategory) {
+		resetNavigator(currentCategory);
 	}
-	if (newCategory !== '') {
-		getPostsByCategory();
+	if (currentCategory !== '') {
 		commentNav().buttons.style.display = 'block';
+		enableControls();
```

Comment:
Some change along the way prevented the navigation buttons from activating when IAmA mode was activated quietly.